app/controllers/concerns/course/lesson_plan/personalization_concern.rb

```diff
@@ -158,12 +170,12 @@ def retrieve_submissions(course_user)
   # NOTE: It is completely possible for negative results (even -infinity), i.e. student needs to go back in time in
   # order to have any hope of completing the course within the limits. The algorithm needs to take care of this.
   def compute_learning_rate_effective_limits(
-    course_user, items, submitted_ids, min_overall_learning_rate, max_overall_learning_rate
+    course_user, items, submissions, min_overall_learning_rate, max_overall_learning_rate
   )
     course_start = items.first.start_at
     course_end = items.last.start_at
     last_submitted_item =
-      items.reverse_each.lazy.select { |item| item.affects_personal_times? && item.id.in?(submitted_ids) }.first
+      items.reverse_each.lazy.select { |item| item.affects_personal_times? && item.id.in?(submissions.keys) }.first
```

Comment:
`submissions` can be an empty array

Reply:
Just pulled `personal-times-playground` directly, can no longer reproduce

Reply:
From eyeballing the `compute_learning_rate_effective_limits`, even if there are no submissions it should be an empty hash (hash merge with hash merge with hash), so I'm not sure why you're getting an array. Might have just been a hot-reloading fluke 😂

Lmk if you can repro again and I'll look into it.